src/main/java/com/google/sps/servlets/movie/MovieDetailsServlet.java

```
@@ -0,0 +1,68 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * @author chris
+ * date: 6/21/2020
+ */
+@WebServlet("/movies/details")
+public class MovieDetailsServlet extends HttpServlet {
+
+    private TmdbMovies moviesQuery = new TmdbMovies(new TmdbApi(KeyConfig.MOVIE_KEY));
+    private Gson gson = new Gson();
+
+    /**
+     * the doGet() for this servlet will return the user movie details provided a movie id and get
+     * movie meta data, if no id is present in the request, the servlet will return a 400. if the query
+     * can't find the movie, it'll return a 500.
+     * @param request: needs a movie id query parameter
+     * @param response: returns a MovieDb object, including movie metadata.
+     * @throws IOException
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json; charset=utf-8");
+
+        Integer id = null;
+
+        if(null != request.getParameter("id")) {
+            id = parseInt(request.getParameter("id"));
+            if(null == id) {
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+        } else {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        // Uses null to default language to en-US
+        MovieDb queryResponse = moviesQuery.getMovie(id, null);
+
+        response.getWriter().println(gson.toJsonTree(queryResponse));
+    }
+
+    /**
+     * parseInt method is used to parse an integer from a string (from a url param)
+     * @param urlParam: the url param to convert from string -> integer
+     * @return either an integer if the string is a valid integer, or null if it can't be parsed
+     */
+    private Integer parseInt(String urlParam) {
+        try {
+            return Integer.parseInt(urlParam);
+        } catch(NumberFormatException e) {
```

What if

---

```
        try {
            return Integer.parseInt(urlParam);
        } catch(Exception e) {
            return null;
        }
```

So were you looking for something like this? I looked through `Integer.parseInt()` and I couldn't find any other instances from an error thrown. The method signature is: 
`    public static int parseUnsignedInt(String s, int radix)
                throws NumberFormatException`

---

Yes, this makes more sense as "all" exceptions are being handled the same way.

---

https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#parseInt(java.lang.String)
I believe this particular function can only ever throw NumberFormatException. My understanding is that functions have to declare all exceptions they could throw, so I'm pretty sure catching NumberFormatException is more specific, and will never crash. Correct me if I'm wrong.

---

Normally, you don't catch "specific" exceptions. You only catch "specific" exceptions, if you have a "specific" action for that type of exception. Or you just catch all exceptions apply the default handler.

---

Good point, you're right.

---

Fixed with cc10931